api/config/v1alpha1/envoyproxy_types.go

```diff
@@ -43,6 +43,15 @@ type EnvoyProxySpec struct {
 	//
 	// +kubebuilder:default={level: {system: info}}
 	Logging ProxyLogging `json:"logging,omitempty"`
+
+	// Bootstrap defines the Envoy Bootstrap as a YAML string.
+	// Visit https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto#envoy-v3-api-msg-config-bootstrap-v3-bootstrap
+	// to learn more about the syntax.
+	// Some fields within the Bootstrap that are required to communicate with the xDS Server (Envoy Gateway) and receive xDS resources
+	// from it are not configurable and will result in the `EnvoyProxy` resource being rejected.
+	// We strongly recommend using `egctl x translate` to generate a `EnvoyProxy` resource with the `Bootstrap` field set to the default
+	// Bootstrap configuration used. You can edit this configuration, and rerun `egctl x translate` to ensure there are no validation errors.
+	Bootstrap string `json:"bootstrap,omitempty"`
```

Comment:
IMO, a string is easy to use, but hard to validate.

Reply:
we will need to unmarshal it to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto#bootstrap-proto and run the object's `Validate` along with any other custom validations we want to do

Reply:
if there are failures, we will highlight those in the `Status` field of `EnvoyProxy` resource

Reply:
this was raised in the EG Community meeting today. Even if we used a strongly typed field here such as envoy's `v3.Bootstrap`, we would need to build extra tooling to convert the imported structure into CRD schema validations which would help validating early and synchronously during a `kubectl apply`.

Reply:
prefer to use a good webhook, not status to tell user what's wrong, not strong option.

Reply:
I tried to use v3.Bootstrap but `controller-gen` was unable to generate a CRD for it.

Reply:
`v3.Bootstrap` came from protobuf, there's a lot of magic.

Reply:
yes, it was decided early on to stick to `controller-gen` tooling rather than protobuf tooling ( which would have helped with this issue using https://pkg.go.dev/istio.io/tools/cmd/cue-gen)

Reply:
https://github.com/envoyproxy/gateway/issues/50 has more details @zirain